doc: Enhance rule documentation (#50)
salesforce/eslint-plugin-lwc: docs/rules/no-unexpected-wire-adapter-usages.md

## Enforce wire adapters to be used with `wire` decorator (no-unexpected-wire-adapter-usages)

> With the [wire reform](https://rfcs.lwc.dev/rfcs/lwc/0000-wire-reform), all the wire adapters can now be invoked programmatically. Ensuring that wire adapters are consumed via the `wire` decorator ensures that their consumption is statically analyzable.

Reviewer: ```suggestion
Due to the [wire reform](https://rfcs.lwc.dev/rfcs/lwc/0000-wire-reform), all wire adapters can be invoked programmatically. Ensuring that wire adapters are consumed via the `wire` decorator ensures that their consumption is statically analyzable.
```
